core/trie/node.go

```diff
@@ -88,30 +88,33 @@ func (n *Node) MarshalBinary() ([]byte, error) {
 
 // UnmarshalBinary deserializes a [Node] from a byte array
 func (n *Node) UnmarshalBinary(data []byte) error {
-	// TODO: Implement and test the following edge cases:
-	//	- Unmarshalling a node with multiple left and right children.
-	//		Currently the assumption is that the node will only have at max 1 left and/or right
-	//		child. However if the UnmarshalBinary is called with multiple left and/or right child
-	//		in any order then UnmarshalBinary will still succeed.
 	if len(data) < felt.Bytes {
 		return ErrMalformedNode{"size of input data is less than felt size"}
 	}
 	n.value = new(felt.Felt).SetBytes(data[:felt.Bytes])
 	data = data[felt.Bytes:]
 
 	stream := bytes.NewReader(data)
+
 	for stream.Len() > 0 {
 		head, err := stream.ReadByte()
 		if err != nil {
 			return err
 		}
-
 		var pathP **bitset.BitSet
 		switch head {
 		case 'l':
-			pathP = &(n.left)
+			if n.left == nil {
```

Comment:
Actually, you can move this check after the `switch` and do it only once.

something like below should work
``` golang
if *pathP != nil {
    return err
}
```

Reply:
Hi, @omerfirmak , your suggestion seems not work in this context. As `*pathP` is point to one of the Node's child,  after enter in case `l` or `r`, it's reference can be decided.

Reply:
It should work. Did you try it, can I see your code?

Reply:
Hi, @omerfirmak, I try to follow your mind, but the code still not work.
Please check [my implement ](https://github.com/liusanchuan/juno/tree/test-pathP-efficient-check).

Reply:
Solved in the efficient way as you suggested, as [here](https://github.com/NethermindEth/juno/pull/508/commits/d48cbf3ea4f457fbfddc8468478e5e99dc7803b7)